StitchCore-iOS/StitchCore-iOS/Services/Internal/StitchServiceImpl.swift

```diff
@@ -13,6 +24,19 @@ internal final class StitchServiceImpl: CoreStitchService, StitchService {
         super.init(requestClient: requestClient, routes: routes, name: name)
     }
 
+    /**
+     * Calls the function for this service with the provided name and arguments.
+     *
+     * - parameters:
+     *     - withName: The name of the function to be called.
+     *     - withArgs: The `BSONArray` of arguments to be provided to the function.
+     *     - completionHandler: The completion handler to call when the function call is complete.
+     *                          This handler is executed on a non-main global `DispatchQueue`.
+     *     - result: The result of the function call as an `Any`, or `nil` if the function call failed.
+     *     - error: An error object that indicates why the function call failed, or `nil` if the function call was
+     *              successful.
+     *
+     */
     public func callFunction(withName name: String,
                              withArgs args: BSONArray,
```

Comment:
[aside] Won't this be onerous for callers to construct arrays rather than a variadic list of some sort?

Reply:
I would consider a variadic list of ExtendedJSONRepresentables... we can also make the BSONArray class better.

Reply:
I just followed what we did in Java. Usage doesn't seem that bad:

```swift
stitchClient.callFunction(withName: "echoArg", withArgs: ["Hello world!"]) { (value, error) in
     // do stuff
}
```

But I'm okay with it if we can do the same in Java, and we can do it without messing with the completion handler pattern we've been using. (If the variadic parameter has to be the last parameter, then it would make the completion handler for callFunction different than every other asynchronous action.